test/e2e_node/gpu_device_plugin.go

```diff
@@ -111,7 +113,9 @@ var _ = framework.KubeDescribe("NVIDIA GPU Device Plugin [Feature:GPUDevicePlugi
 			f.PodClient().Delete(devicePluginPod.Name, &metav1.DeleteOptions{})
 			By("Waiting for GPUs to become unavailable on the local node")
 			Eventually(func() bool {
-				return framework.NumberOfNVIDIAGPUs(getLocalNode(f)) <= 0
```

Comment:
Why did we change this?

Reply:
Because we restart kubelet earlier, calling getLocalNode(f) here could fail with node not ready. Since we just want to check that gpus become unavailable here, I figured it is better to just check the node without needing to wait for it to be ready.

Reply:
Is it possible that the gpus become available only when the node is ready? If so, we would miss that condition here?

Reply:
We don't plan to have such dependency. I think people might have talked about the other way of dependency, which I think we should also avoid to support truly dynamic resource lifecycle.